src/frontend/qt_sdl/main.cpp

```diff
@@ -1412,15 +1416,56 @@ QString MainWindow::loadErrorStr(int error)
     }
 }
 
+std::string extractROM(char* zipName, std::string zipDir){
+    //Open the ZIP archive
+    int err = 0;
+    zip *z = zip_open(zipName, 0, &err);
+    
+    struct zip_stat st;
+    zip_stat_init(&st);
+    zip_stat_index(z, 0, 0, &st); //Get information about the file at index 0
+    char newName[255];
+    strcpy(newName, st.name); //fix for Linux invalid encoding filename	
+    //Allocate memory for its uncompressed contents
+    u8 *contents = new u8[st.size];
```

Comment:
I'd suggest using `auto contents = std::make_unique<u8[]>(st.size)` to avoid potential for memory leaks. Use `contents.get()` instead of `contents` in following code and also include `<memory>`.

Reply:
Thanks, have included this in the libarchive extract function.